crates/cext/src/circuit_library/iqp.rs

```diff
@@ -0,0 +1,109 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use ndarray::ArrayView2;
+use qiskit_circuit::{circuit_data::CircuitData, operations::Param};
+use qiskit_circuit_library::iqp::{check_symmetric, iqp, py_random_iqp};
+
+/// Generate an IQP circuit from an integer interaction matrix.
```

Comment:
This needs the circuit library tag for the docs (same for the random-version of this function below 🙂 )
```suggestion
/// @ingroup QkCircuitLibrary
/// Generate an IQP circuit from an integer interaction matrix.
```

Reply:
Done 😄